### PR TITLE
feat: created an endpoint to add new product

### PR DESCRIPTION
**Pull request by Juninhotech**
# Description
Create an api endpoint to create new product

Link to issue #84 

## What were you told to do?
To create an api endpoint that can allow new product to be created

## What did you do?
1. Created an endpoint
2. Added necessary model and dtos
3. tested the endpoint 

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

### CODE REVIEW

File: src/Hng.Web/Controllers/ProductController.cs [hunk 1]
```diff
@@ -14,11 +15,20 @@ namespace Hng.Web.Controllers
...
         private readonly IMediator _mediator;
+
         public ProductController(IMediator mediator)
         {
             _mediator = mediator;
         }
 
+        [HttpPost]
+        [ProducesResponseType(typeof(ProductDto), StatusCodes.Status201Created)]
+        public async Task<ActionResult<UserDto>> CreateProduct([FromBody] ProductCreationDto body)
```
Comment (Godhanded) [inline src/Hng.Web/Controllers/ProductController.cs:26]: ProductDto . your function return is set to UserDto

File: src/Hng.Web/Controllers/ProductController.cs [hunk 2]
```diff
@@ -34,13 +44,13 @@ public async Task<ActionResult<ProductDto>> GetProductById(Guid id)
 
             var query = new GetProductByIdQuery(id);
             var product = await _mediator.Send(query);
-
             return Ok(new
```
Comment (Godhanded) [inline src/Hng.Web/Controllers/ProductController.cs:47]: Theres a generic response dto in the shared foler in hng.application. return tyhat dto with your product dto set to the data field

**Review by Godhanded — CHANGES_REQUESTED** [review]
Small change and you should be good to go

File: src/Hng.Application/Features/Products/Handlers/CreateProductHandler.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            _mapper = mapper;
+        }
+
+        public async Task<ProductDto> Handle(CreateProductCommand request, CancellationToken cancellationToken)
+        {
+            var product = _mapper.Map<Product>(request.productBody);
+            product.Id = Guid.NewGuid();
+            product.CreatedAt = DateTime.UtcNow;
+            product.UpdatedAt = DateTime.UtcNow;
+            await _repository.AddAsync(product);
```
Comment (chimaihueze) [inline src/Hng.Application/Features/Products/Handlers/CreateProductHandler.cs:31]: After `AddAsync`, you should `SaveChanges` to persist data to the database.

**Review by chimaihueze — CHANGES_REQUESTED** [review]

File: src/Hng.Application/Features/Products/Dtos/ProductDto.cs [hunk 4]
```diff
@@ -13,5 +14,27 @@ public class ProductDto
 
         [JsonPropertyName("description")]
         public string Description { get; set; }
+
+        [JsonPropertyName("category")]
+        public string Category { get; set; }
+
+        public decimal Price { get; set; }
```
Comment (chimaihueze) [inline src/Hng.Application/Features/Products/Dtos/ProductDto.cs:21]: add json property name here too

**Review by chimaihueze — CHANGES_REQUESTED** [review]

**Pull request closed without merging**